sdv/single_table/base.py

```diff
@@ -92,6 +92,94 @@ def __init__(self, metadata, enforce_min_max_values=True, enforce_rounding=True,
+        """Update the transformer parameters."""
+        transformer.locales = self.locales
+        transformer.columns_to_sdtypes = columns_to_sdtypes
+        transformer._list_sdtypes = list_sdtypes
+
+    def _get_address_transformer_parameters(self, column_names):
+        """Get the parameters for the address transformer."""
+        existing_address = {
+            col for col_tuple in self._data_processor.columns_to_mutli_col_transformer
+            for col in col_tuple
+        }
+        columns_to_sdtypes = {}
+        list_sdtypes = []
+
+        for column in column_names:
+            self._check_address_column(column, existing_address)
+            sdtype = self.metadata.columns[column]['sdtype']
+            columns_to_sdtypes[column] = sdtype
+            list_sdtypes.append(sdtype)
+
+        return columns_to_sdtypes, list_sdtypes
+
+    def set_address_columns(self, column_names, anonymization_level='full'):
+        """Set the address multi-column transformer.
+
+        Args:
+            column_names (tuple[str]):
+                The column names to be used for the address transformer.
+            anonymization_level (str):
+                The anonymization level to use for the address transformer.
+        """
+        if self._fitted:
+            warnings.warn(
+                'Please refit your synthesizer for the address changes to appear in'
+                ' your synthetic data.'
+            )
+
+        if anonymization_level not in {'full', 'street_address'}:
+            raise ValueError(
+                f"Invalid value '{anonymization_level}' for parameter 'anonymization_level'."
+                " Please provide 'full' or 'street_address'."
+            )
+
+        if not isinstance(column_names, tuple):
+            column_names = tuple(column_names)
+
+        is_existing_set = any(
+            set(column_names) == set(existing_address)
+            for existing_address in self._data_processor.columns_to_mutli_col_transformer
+        )
+
+        if not is_existing_set:
+            columns_to_sdtypes, sdtypes = self._get_address_transformer_parameters(column_names)
+
+        if anonymization_level == 'full':
+            transformer = self._data_processor._multi_column_transformers.get('address_full')
+        elif anonymization_level == 'street_address':
+            transformer = self._data_processor._multi_column_transformers.get('address_street')
+
+        if transformer is None:
+            warnings.warn(
+                'Setting multi-column address data is a premium feature. '
+                'Please contact Datacebo Inc. for more information.'
+            )
+        else:
+            self._update_address_transformer(transformer, columns_to_sdtypes, sdtypes)
+            transformer._validate_sdtypes()
+
+            self._data_processor.columns_to_mutli_col_transformer[column_names] = transformer
```

Comment:
Move all this logic in the `data_processor` since in most of the cases you are accessing the `self._data_processor` and it's attributes.
At the end of the day the transformers live inside the `data_processor` and not in the `synthesizer`. The `synthesizer` should only be a layer between the user and the `data_processor`.
This may require you to move some of the methods for validation there as well, and pass a locales as parameter to the method.

Reply:
Yes this is what I'm doing today

Reply:
I agree with this